apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -0,0 +1,481 @@
+)
+
+// StagedUpdateRunConditionType identifies a specific condition of the StagedUpdateRun.
+// +enum
+type StagedUpdateRunConditionType string
+
+const (
+	// StagedUpdateRunConditionInitialized indicates whether the staged update run is initialized which means it
+	// has computed all the stages according to the referenced strategy and is ready to start the update.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is initialized.
+	// - "False" means the staged update run encountered an error during initialization.
+	StagedUpdateRunConditionInitialized StagedUpdateRunConditionType = "Initialized"
+
+	// StagedUpdateRunConditionRunning indicates whether the staged update run is proceeding normally.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is proceeding normally.
+	// - "False" means the staged update run is not proceeding normally.
+	// - "Unknown" means it is unknown.
+	StagedUpdateRunConditionRunning StagedUpdateRunConditionType = "Running"
+
+	// StagedUpdateRunConditionWaitingBetweenStage indicates whether the staged update run is waiting between stages.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is waiting between stages.
+	// - "False" means the staged update run is not waiting between stages.
+	// - "Unknown" means it is unknown.
+	StagedUpdateRunConditionWaitingBetweenStage StagedUpdateRunConditionType = "WaitingBetweenStage"
+
+	// StagedUpdateRunConditionCompleted indicates whether the staged update run is completed.
+	// Its condition status can be one of the following:
+	// - "True" means the staged update run is completed successfully.
+	// - "False" means the staged update run encountered an error during running.
+	StagedUpdateRunConditionCompleted StagedUpdateRunConditionType = "Completed"
+)
+
+// StagedUpdateRunList contains a list of StagedUpdateRun.
+// +kubebuilder:resource:scope="Namespaced"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type StagedUpdateRunList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StagedUpdateRun `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope="Namespaced",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageApprovalRequest defines a request for the approval of the stage in the update.
+// Each snapshot MUST have the following labels:
+//   - `ParentStagedUpdateRun` which points to its owner stage update.
+//   - `TargetStage` which is the name of the stage that this approval request is for.
+//   - `IsLatestUpdateRunApproval` which indicates whether this approval request is the latest one related to this update run.
+type StageApprovalRequest struct {
```

Comment:
I would recommend to use ApprovalRequest instead of StageApprovalRequest.

Add TargetStep in the spec. A step can be a stage, a group or a cluster.

```
StepReference {
  RunName
  StageName
  (future) GroupName
  (future) ClusterName
}
```